### DISCUSSION TITLE
clang tidy doesnt recognize "hicpp-avoid-goto"

### DISCUSSION THREAD

**Discussion by SujjithS** [clangd/clangd#1799] [Q&A]
```
#include <iostream>

int main() {
    int x = 5;
    int y = 10;

    // Define a label for goto
    goto myLabel;

    // This variable is unused
    int unusedVariable;

    // This code is unreachable
    std::cout << "This line is never reached." << std::endl;

myLabel:
    // Using x and y to avoid unused variable warnings
    std::cout << "x + y = " << (x + y) << std::endl;

    return 0;
}
```

In the above code clangd displays warning for unusedVariable as follows
```
Code will never be executedclang(-Wunreachable-code)
Unused variable 'unusedVariable'clang(-Wunused-variable)
variable unusedVariable
Type: int
This variable is unused

// In main
int unusedVariable
```

Ideally clang-tidy should recognize the goto and throw warnings since "hicpp-avoid-goto" is included in the .clang-tidy file along with other standards such as [hicpp-no-array-decay, hicpp-signed-bitwise].
I could also see there are some warnings related to hicpp-braces-around-statements which I did not even define anywhere to look for.

Could someone please help me with the hicpp-avoid-goto unrecognized issue?

**Comment by HighCommander4** [comment]
I get a warning about goto in a project with a source file containing the above code, and a `.clang-tidy` file containing `Checks: hicpp-avoid-goto`.

A few questions to help diagnose why it's not working for you:
 * What clangd version are you using
 * Where is your `.clang-tidy` file located and what does it contain?
 * Do you have a `.clangd` file that might be overriding the list of clang-tidy checks that clangd runs with https://clangd.llvm.org/config.html#clangtidy?
 * Can you attach [clangd logs](https://clangd.llvm.org/troubleshooting#gathering-logs)?

**Reply by SujjithS** [reply]
- clangd --version
Ubuntu clangd version 12.0.0-3ubuntu1~20.04.5

- File location
Software/components/component1/
component1 folder consist of inc, src, cMakeList.txt, .clang-tidy, .clangd
.clang-tidy file contents

Checks: '
    # MISRACPP Rule 5-2-5:
      cppcoreguidelines-pro-type-const-cast,
    # MISRACPP Rule 5-2-12:
      hicpp-no-array-decay,
    # MISRACPP Rule 5-3-3:
      google-runtime-operator,
    # MISRACPP Rule 6-4-3:
      hicpp-multiway-paths-covered,
    # MISRACPP Rule 7-3-4:
      google-build-using-namespace, 
    # IVS 2.2.5:
      hicpp-no-malloc,
    # IVS 4.0.6:
      hicpp-avoid-goto,
    # IVS 6.4.1:
      cppcoreguidelines-init-variables,
    # IVS 6.4.5:
      hicpp-avoid-c-arrays,
   '
- I am not sure how to check if it is overriding but .clangd file is also located at the same location where .clang-tidy is and also we make use of a ClangTidy.cmake which consists
```
find_program(CLANG-TIDY_PATH clang-tidy REQUIRED)

function(AddClangTidy target)

  set_target_properties(${target}
    PROPERTIES CXX_CLANG_TIDY
    "${CLANG-TIDY_PATH};-checks=*,\
-llvm-include-order,\
-modernize-use-trailing-return-type,\
-readability-convert-member-functions-to-static"
  )

endfunction()

we invoke AddClangTidy from a cMakeList kept inside component1 as follows

if (BL_ENABLE_CLANG_TIDY_CHECKS)    
    include(ClangTidy)
    AddClangTidy(${LIBRARY_NAME} )
endif()
```

**Reply by SujjithS** [reply]
[logs.txt](https://github.com/clangd/clangd/files/13067190/logs.txt)

logs attached

**Reply by SujjithS** [reply]
I would like to rephrase my statement and be more specific. I am able to see the warnings in the console after building but doesnt show the yellow underlined warnings on the script.(similar to clangd)
